packages/cli/src/scripts/interfaces.ts

```diff
@@ -68,10 +68,10 @@ export type Params =
   | CompileParams;
 
 export interface CreateParams extends Proxy {
-  salt?: string;
-  signature?: string;
-  admin?: string;
-  kind?: ProxyType;
+  salt: string;
+  signature: string;
+  admin: string;
+  kind: ProxyType;
 }
```

Comment:
This change reminded me of [this talk](https://www.youtube.com/watch?v=YR5WdGrpoug), which I coincidentally watched yesterday. TLDR: an object does not have optional fields, it has fields which may be optional depending on the context.

Reply:
Particularly in TypeScript it makes no sense to have a type with all the fields optional because then `Partial<NonOptional>` gives you an optional version of it.